envoy/api/v2/route/route.proto

```diff
@@ -564,6 +564,15 @@ message RedirectAction {
 
   // [#not-implemented-hide:] The scheme portion of the URL will be swapped with "https".
   bool https_redirect = 4;
+
+  // [#not-implemented-hide:] Indicates that during redirection, the matched prefix (or path) should
+  // be swapped with this value. This option allows redirect URLs be dynamically created based on
+  // the request.
+  string prefix_rewrite = 5;
+
+  // [#not-implemented-hide:] Indicates that during redirection, the query portion of the URL will
+  // be removed. Works with *prefix_rewrite* and default value is false.
```

Comment:
Does this only work `prefix_rewrite`, or are you suggesting both can be used at the same time?

Reply:
Initially I was planning to make it work only with `prefix_rewrite`. But on second thoughts I feel it can be an isolated feature & can be used for cases like `host_redirect` & `strip_query`. Will update the wordings accordingly.